gooddata-sdk/gooddata_sdk/catalog/data_source/service.py

```diff
@@ -148,3 +155,58 @@ def load_and_put_declarative_data_sources(
     ) -> None:
         data_sources = self.load_declarative_data_sources(layout_root_path)
         self.put_declarative_data_sources(data_sources, credentials_path, test_data_sources)
+
+    def get_declarative_pdm(self, data_source_id: str) -> CatalogDeclarativeTables:
+        return CatalogDeclarativeTables.from_api(self._layout_api.get_pdm_layout(data_source_id).get("pdm"))
+
+    def put_declarative_pdm(self, data_source_id: str, declarative_tables: CatalogDeclarativeTables) -> None:
+        declarative_pdm = DeclarativePdm(pdm=declarative_tables.to_api())
+        self._layout_api.set_pdm_layout(data_source_id, declarative_pdm)
+
+    def scan_data_source(
+        self, data_source_id: str, scan_request: CatalogScanModelRequest = CatalogScanModelRequest()
+    ) -> CatalogDeclarativeTables:
+        # TODO - how to process warnings? Return it to developers, so they can process it in their way?
+        scan_response = self._scan_api.scan_pdm(data_source_id, scan_request.to_api())
+        warnings = scan_response.get("warnings")
+        if warnings:
+            print("Scan produced the following warnings: ")
```

Comment:
Printing warnings from library is not a good idea:
- it is not possible to block writing messages to output
- user cannot process warning automatically

Lets return raw warnings together with CatalogDeclarativeTables for now. We can provide some method to print data, but it must be called explicitly.

Reply:
warnings printed only on demand.
ScanResult is introduced as a standard Catalog class containing both PDM and warnings.